src/java.base/share/classes/java/io/ClassCache.java

```diff
@@ -87,9 +84,8 @@ T get(Class<?> cl) {
             // We might still have strong referent, and can return it.
             // This guarantees progress for at least one thread on every CacheRef.
             // Clear the strong referent before returning to make the cache soft.
-            T strongVal = ref.getStrong();
+            T strongVal = ref.getAndClearStrong();
```

Comment:
This is a cache, so IIRC, the prevailing case is when CacheRef already exists in the map, and its `strong` is already `null`. So this change introduces an atomic op, which is mostly useless on this path. So we pay a latency price on most `get()`-s without much benefit.